### DISCUSSION TITLE
Use monospaced font for Markdown/file editor

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#4614] [Other Feature Feedback, Questions, & Ideas]
When editing markdown files do I frequently use the table markdown format to create tables.
The issue here is, that certain characters within the font used for the "Write" block do not seem to be truly monospaced which is quite annoying when you try to create a table that is good to look at in its raw form.

Take this table for example:
```markdown
| A   | random | table |
| --- | ------ | ----- |
| Foo | Bar    | Test  |
```
It looks fine in the code block right? Well, the issue is, that the text is completely misaligned within the editor's "Write" field since it doesn't seem to use a monospaced font there.

![image](https://user-images.githubusercontent.com/11576465/125322202-cacee900-e33d-11eb-9f62-8dda376d82c0.png)

This is also a prominent case when using emojis within the table as their width seems to be non-standard and varies on factors like the OS you use and how much their actual width is (See #4308 for a more detailed discussion about issues with Emojis).

Not having a monospaced font here is a real headache when it comes down to creating appealing markdown files. I would then either need to use an external editor on my PC (i.e. Notepad++) or an external site like hastebin to temporary create the markdown I want.

Also, to my knowledge is the raw view (raw.githubusercontent.com) not affected by this and properly monospaced.

**Comment by davidstaheli** [comment]
@Andre601, thank you for posting this! I've wished for the same thing. One approach to this is keeping the current variable-width typography for people who find it easier to read, but also supporting fixed-width typography for the common reasons you wrote above. To satisfy both audiences, what would you think about an icon on the toolbar for toggling between these two?  Would it meet your needs, and is there anything you'd do differently?  Here's a rough mock-up of what it could look like:

![image](https://user-images.githubusercontent.com/1767415/131712047-4466ea68-5214-4d70-b8f3-2018d81e21a8.png)

**Reply by Andre601** [reply]
Not a bad idea and good if it would stay persistent across page reloads.

Otherwise would I think a general appearance setting similar to the theme one would be a more beneficial option.

**Reply by achimmihca** [reply]
I am looking for the same feature. It is not only useful for tables. I want to illustrate stuff in a simple ascii "image".
For example, preemption as blocks on a timeline:
```
             |_second task_|
|_first task_|             |_first task_|
```

**Comment by john-light** [comment]
Does the issue described by OP explain why my table looks fine when I'm editing the file but looks messed up when I view the raw file? Or is this a separate issue that I should create a new discussion for?

![Screenshot 2022-04-18 131129](https://user-images.githubusercontent.com/9424721/163846495-919dee2c-3570-4cdf-abfd-f199aa271d44.png)

![Screenshot 2022-04-18 131147](https://user-images.githubusercontent.com/9424721/163846508-2ce7b7fb-cc07-4cc0-b07b-c7159fdb48fa.png)

**Reply by wjandrea** [reply]
I bet you're using tabs. Newer systems generally render tabs at 4 spaces while older systems generally render them at 8 spaces, so I guess the editor is using 4 and the browser raw view is using 8.

**Comment by BenFrantzDale** [comment]
https://github.blog/changelog/2021-10-12-fixed-width-font-support-in-markdown-enabled-fields/ works for me(!) Why is this not the default?!

**Reply by Abdull** [reply]
Did GitHub make a change very recently that suddenly several users (including you and me and everyone reacting on your helpful comment) unintentionally had their text editor settings changed from monospace to variable-width?

**Reply by BenFrantzDale** [reply]
I don’t know. I’ve always had variable-width fonts in comment boxes.
